hartsfield/api/datahub_archive_url_fetch.py

```diff
@@ -22,79 +22,72 @@
+from hartsfield.api.auth_helper import authorzied_user_required
+from hartsfield.lib.http import tolerant_jsonify
+import hartsfield.api.read_aws_secret
 
 PUBLIC_CONFIGS = [
     'DEV_AUTH_ENABLED',
     'HARTSFIELD_ENV',
     'TIMEZONE',
 ]
 
-gcp_json_credentials = app.config['GCP_JSON_CREDENTIALS']
-gcp_json_credentials_dict = json.loads(gcp_json_credentials)
+AWS_SECRETS_NAME_GCP_JSON_CREDENTIALS = app.config['AWS_SECRETS_NAME_GCP_JSON_CREDENTIALS']
 
-# TODO: pass in gs url as input value to @app.route('/api/fetch_url_direct') from form user front-end form submission
-gs_source_url="gs://ucb-datahub-archived-homedirs/spring-2021/datahub.berkeley.edu/peterphu-2edo.tar.gz"
-# This will probably be request.args['gs_source_url'] in the def block...but that whole "request" business needs to be brought in etc.
 
-@app.route('/api/fetch_url_direct')
+gcp_json_credentials_from_aws = hartsfield.api.read_aws_secret.read_aws_secret(AWS_SECRETS_NAME_GCP_JSON_CREDENTIALS)
+gcp_json_credentials_dict = json.loads(gcp_json_credentials_from_aws)
+
+@app.route('/api/fetch_url_direct', methods=['POST'])
+@authorzied_user_required
 def fetch_url_direct():
 
-    # parse the input gs url to get bucket and blob names
-    bucket_and_blob_string = gs_source_url.replace("gs://", "")
-    bucket_and_blob_list = bucket_and_blob_string.split("/")
+    params = request.get_json()
+    gs_source_url = params.get('gs_source_url')
+    gs_source_url = gs_source_url.strip()
+    if not re.match(r'gs://.{3,}/.+', gs_source_url):
+        error_message = 'The submitted data \"' + gs_source_url + '\" is not a valid gs_source_url.'
+        v = {'response': error_message, 'status': 'error'}
+        return tolerant_jsonify(v)
+
+    bucket_and_blob_string = gs_source_url.replace('gs://', '')
+    bucket_and_blob_list = bucket_and_blob_string.split('/')
     bucket_name = bucket_and_blob_list.pop(0)
-    blob_name = "/".join(bucket_and_blob_list)
+    blob_name = '/'.join(bucket_and_blob_list)
 
     # instantiate gcp storage client plus with bucket and blob objects
     credentials = service_account.Credentials.from_service_account_info(gcp_json_credentials_dict)
     storage_client = storage.Client(project=gcp_json_credentials_dict['project_id'], credentials=credentials)
     bucket = storage_client.bucket(bucket_name)
     blob = bucket.blob(blob_name)
 
-    # do some checks to confirm that the bucket and blob exist
     try:
         stats = storage.Blob(bucket=bucket, name=blob_name).exists(storage_client)
     except Exception as e:
-        error_message = "There was a problem trying to get stats on the requested blob \"" + blob_name + "\" in the requested bucket \"" + bucket_name +"\":\n\n " + str(e)
+        error_message = 'There was an exception trying to do the GCP storage operation with the submitted data \"' + \
+                        gs_source_url + \
+                        '\'. When GCP tried, it told us: \"' + \
+                        str(e) + '\"'
```

Comment:
Minor code style concern: Use f-string with triple quotes when string is multi-line.
```
m = f"""
     Hello world. My name is {name}.
"""
```